bundles/org.openhab.binding.mybmw/src/main/java/org/openhab/binding/mybmw/internal/handler/MyBMWBridgeHandler.java

```diff
@@ -0,0 +1,157 @@
+import org.openhab.binding.mybmw.internal.utils.BimmerConstants;
+import org.openhab.binding.mybmw.internal.utils.Constants;
+import org.openhab.binding.mybmw.internal.utils.Converter;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MyBMWBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class MyBMWBridgeHandler extends BaseBridgeHandler implements StringResponseCallback {
+    private final Logger logger = LoggerFactory.getLogger(MyBMWBridgeHandler.class);
+    private HttpClientFactory httpClientFactory;
+    private Optional<VehicleDiscovery> discoveryService = Optional.empty();
+    private Optional<MyBMWProxy> proxy = Optional.empty();
+    private Optional<ScheduledFuture<?>> initializerJob = Optional.empty();
+    private Optional<String> troubleshootFingerprint = Optional.empty();
+    private String localeLanguage;
+
+    public MyBMWBridgeHandler(Bridge bridge, HttpClientFactory hcf, String language) {
+        super(bridge);
+        httpClientFactory = hcf;
+        localeLanguage = language;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // no commands available
+    }
+
+    @Override
+    public void initialize() {
+        troubleshootFingerprint = Optional.empty();
+        updateStatus(ThingStatus.UNKNOWN);
+        MyBMWConfiguration config = getConfigAs(MyBMWConfiguration.class);
+        if (config.language.equals(Constants.LANGUAGE_AUTODETECT)) {
+            config.language = localeLanguage;
+        }
+        if (!checkConfiguration(config)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        } else {
+            proxy = Optional.of(new MyBMWProxy(httpClientFactory, config));
+            // give the system some time to create all predefined Vehicles
+            // check with API call if bridge is online
+            initializerJob = Optional.of(scheduler.schedule(this::requestVehicles, 2, TimeUnit.SECONDS));
+            Bridge b = super.getThing();
+            List<Thing> children = b.getThings();
+            logger.debug("Update {} things", children.size());
+            children.forEach(entry -> {
+                ThingHandler th = entry.getHandler();
+                if (th != null) {
+                    th.dispose();
+                    th.initialize();
```

Comment:
What's the purpose of this? Actually the framework initializes the child Things when the bridge is initialized.